biothings/tests/biothings_pytest_plugin/biothings_pytest_plugin.py

```diff
@@ -0,0 +1,246 @@
+            print(" └─ Stored new build version to S3.")
+        except Exception as e:
+            print(f" └─ Failed to store build version in S3. Error: {str(e)}")
+    else:
+        print(" └─ No valid build version found, not storing to S3.")
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_collection_modifyitems(config, items):
+    """Skip all tests if certain conditions are met."""
+    build_version_s3 = fetch_build_version_s3()
+    build_version_hub = fetch_build_version_hub()
+    github_event_name = os.getenv("GITHUB_EVENT_NAME", "")
+    os.environ["SEND_SLACK_NOTIFICATION?"] = "True"
+
+    if build_version_hub == build_version_s3 and github_event_name != "workflow_dispatch":
+        os.environ["SEND_SLACK_NOTIFICATION?"] = "False"
+        print("No need to run the tests.")
+        print(" └─ The S3 and Hub build versions are the same.")
+        # Skip all tests
+        for item in items:
+            item.add_marker(pytest.mark.skip(reason="Skipped due to matching build versions"))
+    else:
+        # Store new build version if tests are going to run
+        store_build_version_s3(build_version_hub)
+
+# Hook to run pytest and send Slack notification
+@pytest.hookimpl(tryfirst=True)
+def pytest_terminal_summary(terminalreporter: TerminalReporter, exitstatus: int, config):
+    """Customize pytest terminal summary and send to Slack."""
+
+    if os.getenv("SEND_SLACK_NOTIFICATION?") == "True":
+        SLACK_WEBHOOK_URL = os.getenv("SLACK_WEBHOOK_URL")
+        SLACK_CHANNEL = os.getenv("SLACK_CHANNEL")
+        SLACK_USERNAME = os.getenv("APPLICATION_NAME")
+
+        # Collect test summary information
+        total_tests = terminalreporter.stats.get('passed', []) + \
+                    terminalreporter.stats.get('failed', []) + \
+                    terminalreporter.stats.get('error', []) + \
+                    terminalreporter.stats.get('skipped', [])
+
+        passed_tests = len(terminalreporter.stats.get('passed', []))
+        failed_tests = len(terminalreporter.stats.get('failed', []))
+        error_tests = len(terminalreporter.stats.get('error', []))
+        skipped_tests = len(terminalreporter.stats.get('skipped', []))
+
+        # Prepare error details
+        error_details = ""
+        for test in terminalreporter.stats.get('failed', []) + terminalreporter.stats.get('error', []):
+            error_details += f"• *Test*: `{test.nodeid}`\n"
+            error_details += f"  _Details:_\n```\n{''.join(test.longreprtext.splitlines(keepends=True)[-10:])}```\n\n"
+        error_details = "*Error Details:*\n" + error_details if error_details else ""
+
+        # Determine status emoji and color
+        status_emoji = ":thumbsup:" if failed_tests == 0 and error_tests == 0 else ":thumbsdown:"
+        bug_emoji = "" if failed_tests == 0 and error_tests == 0 else ":bug-happy:"
+        status_color = "good" if failed_tests == 0 and error_tests == 0 else "danger"
+
+        # Create the payload for Slack
```

Comment:
Given how you've structured this, I'd probably move the code from here to line 208 into a function call something like `post_pytest_summary`. Then you can structure that function like `fetch_build_version_hub` with a try catch schema. Because at the moment that requests could fail due to a timeout to connect to the `SLACK_WEBHOOK_URL` and we wouldn't catch it